flat_mutation_reader.cc

```diff
@@ -433,15 +433,26 @@ flat_mutation_reader
 flat_mutation_reader_from_mutations(reader_permit permit,
                                     std::vector<mutation> ms,
                                     const dht::partition_range& pr,
-                                    const query::partition_slice& slice,
+                                    const query::partition_slice& query_slice,
                                     streamed_mutation::forwarding fwd) {
+    const auto reversed = query_slice.options.contains(query::partition_slice::option::reversed);
+    auto slice = reversed
+            ? query::half_reverse_slice(*ms.front().schema(), query_slice)
```

Comment:
Do we silently assume that
1) the vector `ms` is nonempty
2) all the mutations refer to the same schema?

If we do, it would probably be nice to document this condition.

Reply:
Yes, and this assumption is preexisting. Can still document it though.

Reply:
BTW I think not adding a separate schema parameter to this method was a mistake. I will address this in a follow-up.

Reply:
BTW. this slice unreversing is not necessary I think, `query::clustering_key_filter_ranges::get_ranges` handles half-reversed slices (it unreverses the ranges) already.

(however, I would prefer to get rid of `get_ranges` eventually, if we want to switch to native-reverse slices)

Reply:
I think it is better if we have this here as a reminder that we have to clean this up when we switch to native slice. I agree about dropping `get_ranges()` too, we can do it after the switch.